packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -13,12 +15,56 @@ interface State {
 
 type ActionTypes =
     | CrudGetManyReferenceSuccessAction
+    | CrudDeleteSuccessAction
     | { type: 'OTHER_ACTION'; payload: any; meta?: any };
 
 const oneToManyReducer: Reducer<State> = (
     previousState = initialState,
     action: ActionTypes
 ) => {
+    if (action.meta && action.meta.fetch === DELETE && action.meta.optimistic) {
+        const relatedTo = Object.keys(previousState).filter(key =>
+            key.includes(action.meta.resource)
```

Comment:
How can a `Object.keys` result can be an array.. ? :o

Reply:
Ow. `includes` works on strings too. My bad.

Reply:
?